src/controllers/crate_owner_invitation.rs

```diff
@@ -41,23 +41,46 @@ pub fn handle_invite(req: &mut dyn Request) -> AppResult<Response> {
         serde_json::from_str(&body).map_err(|_| cargo_err("invalid json request"))?;
 
     let crate_invite = crate_invite.crate_owner_invite;
+    let user_id = req.user()?.id;
 
     if crate_invite.accepted {
-        accept_invite(req, conn, crate_invite)
+        accept_invite(req, conn, crate_invite, user_id)
     } else {
         decline_invite(req, conn, crate_invite)
     }
 }
 
+/// Handles the `PUT /me/crate_owner_invitations/accept/:token` route.
+pub fn handle_invite_with_token(req: &mut dyn Request) -> AppResult<Response> {
+    let conn = req.db_conn()?;
+    let req_token = &req.params()["token"];
+
+    let crate_owner_invite: CrateOwnerInvitation = crate_owner_invitations::table
+        .filter(crate_owner_invitations::token.eq(req_token))
+        .first::<CrateOwnerInvitation>(&*conn)?;
+
+    // TODO: Return 404 if no crate_owner_invite was found
```

Comment:
I think the `first`...`?` should return 404 already -- [diesel's `first`](https://docs.diesel.rs/diesel/query_dsl/trait.RunQueryDsl.html#method.first) returns `Err(NotFound)` if no records are returned, and [this code converts `Diesel::NotFound`](https://github.com/rust-lang/crates.io/blob/efeb9d0cef99b8575f97715342794a9518199f69/src/util/errors.rs#L72) to [crates.io's `NotFound`](https://github.com/rust-lang/crates.io/blob/efeb9d0cef99b8575f97715342794a9518199f69/src/util/errors.rs#L232), which [returns 404](https://github.com/rust-lang/crates.io/blob/efeb9d0cef99b8575f97715342794a9518199f69/src/util/errors.rs#L245). That conversion happens as part of the `?` operator.

When I'm on this branch and I visit `http://localhost:4200/accept-invite/foo`, I see the request to `http://localhost:4200/api/v1//me/crate_owner_invitations/accept/foo` get a 404. So I think this TODO is TODONE! :)

Reply:
😁 🎉